src/components/MapView/Layers/AdminLevelDataLayer/index.tsx

```diff
@@ -61,6 +65,16 @@ function AdminLevelDataLayers({ layer }: { layer: AdminLevelDataLayerProps }) {
     return null;
   }
 
+  if (
+    layer.boundary &&
+    !map
+      ?.getStyle()
+      .layers?.map(l => l.id)
+      .includes(`${boundaryId}-line`)
+  ) {
+    return null;
+  }
```

Comment:
I would remove the `layer.boundary &&` condition, since you also want this check to work when you switch back to a normal layer which uses the default admin boundary.

If this returns null, will the data load eventually?

Reply:
Yeah. The data loads eventually. The case for `layer.boundary` is to only do this check for layers with defined `boundary`. Without it normal layers would return null which is the reverse of what we want.

Reply:
You were correct. It is no longer needed since we will always have `boundaryID`